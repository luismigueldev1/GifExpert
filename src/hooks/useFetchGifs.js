import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
export default function useFetchGifs(category) {
  const [state, setState] = useState({
    data: [],
    loading: true,
    error: null,
  });

  useEffect(() => {
    getGifs(category)
      .then((data) => setState({ data: data, loading: false }))
      .catch((error) => setState({ loading: false, error: error }));
  }, [category]);

  return state;
}
