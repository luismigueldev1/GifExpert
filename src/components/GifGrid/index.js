import React from "react";
import useFetchGifs from "../../hooks/useFetchGifs";
import GifGridItem from "../GifGridItem";
import PropTypes from "prop-types";

function GifGrid({ category }) {
  const { data: images, loading, error } = useFetchGifs(category);

  return (
    <div>
      <h3 className="animate__animated animate__fadeIn ">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash animate__infinite">
          Cargando...
        </p>
      )}
      {error && { error }}
      <ol className="card-grid">
        {images.map((image) => {
          return <GifGridItem key={image.id} {...image} />;
        })}
      </ol>
    </div>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
