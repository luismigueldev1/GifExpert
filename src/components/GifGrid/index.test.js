import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from ".";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid/>", () => {
  const category = "Pokemon";

  test("debe mostrarse correctamente", () => {
    useFetchGifs.mockReturnValue({
      loading: true,
      data: [],
      error: null,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes con useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://google.com",
        title: "Imagen de Prueba",
      },
      {
        id: "123",
        url: "https://google.com",
        title: "Imagen de Prueba",
      },
    ];
    useFetchGifs.mockReturnValue({
      loading: false,
      data: gifs,
      error: null,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
