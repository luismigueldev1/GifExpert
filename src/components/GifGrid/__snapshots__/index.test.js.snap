// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn "
  >
    Pokemon
  </h3>
  <ol
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Imagen de Prueba"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Imagen de Prueba"
      url="https://google.com"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifGrid/> debe mostrarse correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn "
  >
    Pokemon
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Cargando...
  </p>
  <ol
    className="card-grid"
  />
</div>
`;
